From 7c309395136300e8e65f0a7706275932ed655bd7 Mon Sep 17 00:00:00 2001
From: Jan Petrous <jan.petrous@nxp.com>
Date: Tue, 17 May 2022 17:49:27 +0200
Subject: [PATCH 021/102] drivers: net: s32-cc: fix eth fixup for RDB2 rev.D

The fixup was restricted to apply only when SGMII
interface is used.

Issue: ALB-7446
Upstream-Status: Pending 

Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/eth.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-s32/s32-cc/eth.c b/arch/arm/mach-s32/s32-cc/eth.c
index 597d7f9e0f..a5b6bd0701 100644
--- a/arch/arm/mach-s32/s32-cc/eth.c
+++ b/arch/arm/mach-s32/s32-cc/eth.c
@@ -135,7 +135,10 @@ static void ft_enet_pfe_fixup_netif(u32 idx, void *fdt)
 		ft_update_eth_addr_by_name("pfe", idx, fdt, nodeoff);
 
 #ifdef CONFIG_NXP_S32GRDB_BOARD
-		ft_enet_pfe_fixup_phy(idx, fdt, nodeoff);
+		/* SGMII PHY address fixup needed by RDB2 rev.D */
+		if (idx == 1 &&
+		    pfeng_cfg_emac_get_interface(idx) == PHY_INTERFACE_MODE_SGMII)
+			ft_enet_pfe_fixup_phy(idx, fdt, nodeoff);
 
 		if (idx == 0)
 			ft_enet_pfe_fixup_fixed_link(idx, fdt, nodeoff);
-- 
2.17.1

